ctera/src/main/java/ch/cyberduck/core/ctera/CteraSession.java

```diff
@@ -178,9 +272,74 @@ public <T> T _getFeature(final Class<T> type) {
         if(type == UrlProvider.class) {
             return (T) new CteraUrlProvider(host);
         }
+        if(type == AttributesFinder.class) {
+            return (T) attributes;
+        }
+        if(type == ListService.class) {
+            return (T) list;
+        }
+        if(type == AclPermission.class) {
+            return (T) new CteraAclPermissionFeature(this);
+        }
+        if(type == Read.class) {
+            return (T) new DAVReadFeature(this) {
```

Comment:
Extract inner class to upper.

Reply:
Done e57ba170367804bdddabca3d79878d9cc6a410ec